internal/dockerignore/ignore_test.go

```diff
@@ -0,0 +1,76 @@
+package dockerignore
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/tilt/internal/ignore"
+	"github.com/windmilleng/tilt/internal/testutils"
+)
+
+func TestIsIgnored(t *testing.T) {
+	tf := newTestFixture(t, "node_modules")
+	tf.AssertResult(tf.JoinPath("node_modules", "foo"), true, false)
+	tf.AssertResult(tf.JoinPath("foo", "bar"), false, false)
+}
+
+func TestComment(t *testing.T) {
+	tf := newTestFixture(t, "# generated code")
+	tf.AssertResult(tf.JoinPath("node_modules", "foo"), false, false)
+	tf.AssertResult(tf.JoinPath("foo", "bar"), false, false)
+}
+
+func TestGlob(t *testing.T) {
+	tf := newTestFixture(t, "*/temp*")
+	tf.AssertResult(tf.JoinPath("somedir", "temporary.txt"), true, false)
+	tf.AssertResult(tf.JoinPath("somedir", "temp"), true, false)
+}
+
+func TestOneCharacterExtension(t *testing.T) {
+	tf := newTestFixture(t, "temp?")
+	tf.AssertResult(tf.JoinPath("tempa"), true, false)
+	tf.AssertResult(tf.JoinPath("tempeh"), false, false)
+	tf.AssertResult(tf.JoinPath("temp"), false, false)
+}
+
+type testFixture struct {
+	repoRoot *testutils.TempDirFixture
```

Comment:
fwiw instead of having this be a named field, if you just do
```
type testFixture struct {
    *testutils.TempDirFixture
    t *testing.T
    ...
}
```
it does something something struct embedding magic that means that a tf has access to all of `TempDirFixture`'s methods on the top level, so you don't need to define `JoinPath` that just calls out to `JoinPath`.